developer/guides/troubleshooting/layout-shift.mdx

```diff
@@ -0,0 +1,182 @@
+            </header>
+
+            {children}
+            
+            <footer className="flex items-center justify-between bg-gray-600 p-4 px-10 text-white">
+              <h1 className="text-3xl font-bold underline">footer</h1>
+            </footer>
+          </Providers>
+        </MakeswiftProvider>
+      </body>
+    </html>
+  )
+}
+```
+</CodeGroup>
+
+Nothing seems wrong with this code, right? Now let's run the app and see what happens.
+
+During the initial load, you will see the header and the footer collapsing together,
+and then, in a moment, a page content appearing between them. Exactly as on the video above.
+
+Congratulations! We have just created a site with a layout shift.
+
+<Note>
+To make the layout shift more noticeable, we added `header` and `footer` components to the layout
+</Note>
+
+## Diagnosing a layout shift
+
+Let's investigate the layout shift we just created. We are going to use **React Profiler** from the
+[React Developer Tools](https://react.dev/learn/react-developer-tools) to see what is happening.
+
+Open **DevTools** in your browser and navigate to the **Profiler** tab. Record a profile of a page reload
+and take a closer look at the updates:
+
+<Frame>
+  <video
+    muted
+    loop
+    playsInline
+    controls
+    title="React profiler"
+    className="w-full aspect-video"
+    src="/images/developer/guides/troubleshooting/layout-shift/react-profiler.mp4"
+  ></video>
+</Frame>
+
+As you can see, the `SessionProvider` is updated twice. The first update corresponds to the
+initial render of the `SessionProvider` component:
+
+<Frame>
+  <img
+    src="/images/developer/guides/troubleshooting/layout-shift/react-profiler-render-1.png"
+    alt="First update to the SessionProvider"
+  />
+</Frame>
+
+The second update corresponds to the re-render of the `SessionProvider` component and is caused by changed props:
+
+<Frame>
+  <img
+    src="/images/developer/guides/troubleshooting/layout-shift/react-profiler-render-2.png"
+    alt="Second update to the SessionProvider"
+  />
+</Frame>
+
+## What's happening?
+
+So what's going on here? Why is this happening?
+
+This is related to the way React handles updates during the hydration process.
+
+Let's take a closer look at the `SessionProvider` [component](https://next-auth.js.org/getting-started/client#sessionprovider).
+It accepts a `session` prop that is used to determine the current session state. The problem arises when the `session` prop is
+updated *during the hydration process*.
+
+Here is the summary of what is going on:
+- The page is server-side rendered (SSR).
+- The server sends the HTML to the client, and React begins hydration.
+- While hydration is in progress, the client side receives a new update, in this case `session` value for the `SessionProvider` component.
+- This update interrupts hydration, causing React to discard the in-progress hydration and re-render the components on the client with the updated session value.
```

Comment:
> We replace the server rendered content with a client side fallback if this suspends on the client. 

[From React team comment](https://github.com/facebook/react/pull/25692#:~:text=We%20replace%20the%20server%20rendered%20content%20with%20a%20client%20side%20fallback%20if%20this%20suspends%20on%20the%20client.).

I think we need to specify that ultimately what's causing the layout shift is that they replace the server rendered content with a client side feedback, which is `null` or empty in this case.